File: Java/Hussain Sir Java/Fundamentals/Language/Inheritance/InterfaceTest1.java
import edu.met.banking.*;

class InterfaceTest1{

	private static void deductTax(Account[] accounts){
		for(Account acc : accounts){
			double extra = acc.getBalance() - 25000;
			if(extra > 0){
				try{
					acc.withdraw(0.05 * extra);
				}catch(InsufficientFundsException e){}
			}
		}
	}

	private static void payAnnualInterest(Account[] accounts){
		for(Account acc : accounts){
			if(acc instanceof Profitable){
				Profitable p = (Profitable)acc;
				double amount = p.interest(12);
				acc.deposit(amount);
			}
		}
	}

	public static void main(String[] args){
		Account[] bank = new Account[5];
		bank[0] = Banker.openSavingsAccount();
		bank[0].deposit(5000);
		bank[1] = Banker.openCurrentAccount();
		bank[1].deposit(20000);
		bank[2] = Banker.openSavingsAccount();
		bank[2].deposit(25000);
		bank[3] = Banker.openCurrentAccount();
		bank[3].deposit(40000);
		bank[4] = Banker.openSavingsAccount();
		bank[4].deposit(45000);
		deductTax(bank);
		payAnnualInterest(bank);
		for(Account acc : bank)
			System.out.printf("%d\t%.2f%n", acc.getId(), acc.getBalance());
	}

}

File: Fundamentals/Language/Inheritance/interfacetest1.cs
using Banking;
using System;

static class Program
{
	private static void DeductTax(Account[] accounts)
	{
		foreach(var acc in accounts)
		{
			double tax = 0.05 * (acc.Balance - 30000);
			if(tax > 0)
				acc.Withdraw(tax);
		}
	}

	private static void PayAnnualInterest(Account[] accounts)
	{
		foreach(Account acc in accounts)
		{
			IProfitable p = acc as IProfitable;
			p?.PayInterest(12); //if(p != null) p.PayInterest(12)	
		}
	}
	public static void Main()
	{
		Account[] bank = new Account[5];
		bank[0] = Banker.OpenSavingsAccount();
		bank[0].Deposit(5000);
		bank[1] = Banker.OpenCurrentAccount();
		bank[1].Deposit(20000);
		bank[2] = Banker.OpenSavingsAccount();
		bank[2].Deposit(25000);
		bank[3] = Banker.OpenCurrentAccount();
		bank[3].Deposit(40000);
		bank[4] = Banker.OpenSavingsAccount();
		bank[4].Deposit(45000);

		DeductTax(bank);
		PayAnnualInterest(bank);

		foreach(var acc in bank)
			Console.WriteLine($"{acc.Id}\t{acc.Balance:0.00}");

	}
}


